app/models/user.rb

```diff
@@ -5,6 +5,10 @@ class User < ApplicationRecord
   has_many :password_reset_tokens, dependent: :destroy
   has_many :sessions, dependent: :destroy
   has_many :events, dependent: :destroy
+  has_many :mentorship_roles_as_mentor, class_name: "Mentorship", foreign_key: "mentor_id"
+  has_many :mentorship_roles_as_applicant, class_name: "Mentorship", foreign_key: "mentor_id"
+  has_many :mentors, through: :mentorship_roles_as_mentor
+  has_many :applicants, through: :mentorship_roles_as_applicant
```

Comment:
First time I've set these types of associations up. Used this as a reference: https://www.spacevatican.org/2008/5/6/creating-multiple-associations-with-the-same-table/ and chatgpt seems to agree!

`mentorship_roles_as_mentor` and `mentorship_roles_as_applicant` seem intuitive enough. Another option I played around with were simply `mentoring` and a`applying` but it didn't seem clear by itself.

Reply:
Had wrong foreign_key on :mentorship_roles_as_applicant. Updated as applicant_id and force pushed to keep it within a single commit.